lib/dfe_sign_in/api/user.rb

```diff
@@ -2,13 +2,57 @@ module DfeSignIn
   module Api
     class User
       include DfeSignIn::Utils
+      extend DfeSignIn::Utils
 
       attr_accessor :organisation_id,
-        :user_id
+        :user_id,
+        :organisation_name,
+        :given_name,
+        :family_name,
+        :email
 
-      def initialize(organisation_id:, user_id:)
-        self.organisation_id = organisation_id
-        self.user_id = user_id
+      class << self
+        def all
+          get_users.map do |u|
+            new(
+              organisation_id: u["organisation"]["id"],
+              organisation_name: u["organisation"]["name"],
+              user_id: u["userId"],
+              given_name: u["givenName"],
+              family_name: u["familyName"],
+              email: u["email"]
+            )
+          end
+        end
+
+        def all_users_endpoint
+          uri = URI(DfeSignIn.configuration.base_url)
+          uri.path = "/users"
+          uri
+        end
+
+        def get_users
+          uri = all_users_endpoint
+          body = get(uri)
+          page_number = 2
+          users = body["users"]
+          while page_number <= body["numberOfPages"]
+            uri.query = "page=#{page_number}"
+            page = get(uri)
+            users << page["users"]
```

Comment:
Would using [`Array#concat`](https://apidock.com/ruby/Array/concat) remove the need for the flatten call further down?

```suggestion
            users.concat page["users"]
```